cg/store/crud/read.py

```diff
@@ -1736,11 +1737,16 @@ def get_all_pools_to_deliver(self) -> list[Pool]:
         )
         return records.all()
 
-    def get_orders(self, limit: int | None = None) -> list[Order]:
+    def get_orders(self, orders_request: OrdersRequest | None = None) -> list[Order]:
         """Returns a list of entries in the table Order."""
         records: Query = self._get_query(table=Order)
-        if limit is not None:
-            records = records.limit(limit)
+        if orders_request:
+            limit: int | None = orders_request.limit
+            workflow: str | None = orders_request.workflow
+            if limit is not None:
+                records = records.limit(limit)
+            if workflow is not None:
+                records = records.filter(Order.workflow == workflow)
```

Comment:
Have a look at the store filters `cg/store/filters/` when writing queries and filter functions that access the store we have a certain schema that we follow.

We define filters with a single responsiblity that are then called in the crud to construct a more complex query.

Reply:
This also looks like it would be two separate queries.

1) `get_orders`
2) `get_orders_for_workflow`